packages/core/src/main/shims/browser/assetCryptography.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {EncryptedAsset} from '../../crypto/root';
+const {crypto} = window;
+
+const equalHashes = (bufferA: Buffer, bufferB: Buffer): boolean => {
+  const arrayA = new Uint32Array(bufferA);
+  const arrayB = new Uint32Array(bufferB);
+  return arrayA.length === arrayB.length && arrayA.every((value, index) => value === arrayB[index]);
+};
+
+export const decryptAsset = async ({
+  cipherText,
+  keyBytes,
+  sha256: referenceSHA256,
+}: EncryptedAsset): Promise<Buffer> => {
+  const computedSHA256 = await crypto.subtle.digest('SHA-256', cipherText);
+
+  if (!equalHashes(new Buffer(computedSHA256), referenceSHA256)) {
```

Comment:
Always consider the positive case first because our brain doesn't like to think in boolean negation.